src/Agent.Listener/Agent.cs

```diff
@@ -10,102 +10,121 @@ namespace Microsoft.VisualStudio.Services.Agent.Listener
     [ServiceLocator(Default = typeof(Agent))]
     public interface IAgent : IAgentService
     {
+        CancellationTokenSource TokenSource { get; set; }
         Task<int> ExecuteCommand(CommandLineParser parser);
     }
 
     public sealed class Agent : AgentService, IAgent
     {
+        public CancellationTokenSource TokenSource { get; set; }
```

Comment:
It is set in Program.MainAsync:
                    using (agent.TokenSource = new CancellationTokenSource())
                    {
                        rc = await agent.ExecuteCommand(parser);
                    }
This property allows for the caller to cancel IAgentExecuteCommand, but currently the token is set only internally in response to CTRL-C. This may change when we add the code to stop the service.